appscale/tools/appscale.py

```diff
@@ -764,12 +757,19 @@ def destroy(self):
     if "EC2_URL" in contents_as_yaml:
       os.environ["EC2_URL"] = contents_as_yaml["EC2_URL"]
 
-    if 'keyname' in contents_as_yaml:
-      command.append("--keyname")
-      command.append(contents_as_yaml['keyname'])
+    if clean:
+      LocalState.ensure_user_wants_to_clean()
+      all_ips = LocalState.get_all_public_ips(keyname)
+      for ip in all_ips:
+        RemoteHelper.ssh(ip, keyname, self.TERMINATE, is_verbose)
```

Comment:
Good point. What would you suggest to do here? the terminate.rb clean is needed to clean the disk. Should I check if clean is set in terminate_virtualized_instances and avoid the call there?